From 7ee6a339af0399e0a1f5ce75bf7064f5476fb1fd Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Wed, 15 Sep 2021 15:40:30 -0700
Subject: [PATCH 12/27] ath12k: hw settings for EHT

Add details in ieee80211_register_hw() for EHT support (yes/no),
and include EHT capabilities maximum length for scan buffer length
calculation.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 net/mac80211/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/mac80211/main.c b/net/mac80211/main.c
index 289b7e835995..9cc2967c4dd4 100644
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -1025,9 +1025,9 @@ int ieee80211_register_hw(struct ieee80211_hw *hw)
 			supp_eht = supp_eht || iftd->eht_cap.has_eht;
 		}
 
-		/* HT, VHT, HE require QoS, thus >= 4 queues */
+		/* HT, VHT, HE, EHT require QoS, thus >= 4 queues */
 		if (WARN_ON(local->hw.queues < IEEE80211_NUM_ACS &&
-			    (supp_ht || supp_vht || supp_he)))
+			    (supp_ht || supp_vht || supp_he || supp_eht)))
 			return -EINVAL;
 
 		/* EHT requires HE support */
-- 
2.31.1

